py/server/deephaven/multijoin.py

```diff
@@ -0,0 +1,126 @@
+#
+#     Copyright (c) 2016-2023 Deephaven Data Labs and Patent Pending
+#
+"""This module exposes the interfaces for efficient multi-table natural join."""
+from typing import Sequence, Union, List
+import jpy
+
+from deephaven import DHError
+from deephaven._wrapper import JObjectWrapper
+from deephaven.table import Table
+from deephaven.jcompat import to_sequence
+from deephaven.update_graph import auto_locking_ctx
+
+_JMultiJoinInput = jpy.get_type("io.deephaven.engine.table.MultiJoinInput")
+_JMultiJoinTable = jpy.get_type("io.deephaven.engine.table.MultiJoinTable")
+_JMultiJoinFactory = jpy.get_type("io.deephaven.engine.table.MultiJoinFactory")
+
+"""
+The call syntax is the following:
+
+# complex join
+mj_input = [
+    MultiJoinInput(table=t1, on="key"), # all columns added
+    MultiJoinInput(table=t2, on="key=otherKey", joins=["col1", "col2"]), #specific columns added
+]
+multitable = multi_join(input=mj_input)
+
+#simple joins
+multitable = multi_join(tables=[t1,t2], on="common_key") # all columns from t1,t2 included in output
+multitable = multi_join(tables=[t1,t2], on=["common_key1", "common_key2"]) # all columns from t1,t2 included
+
+"""
+
+
+class MultiJoinInput(JObjectWrapper):
+    """A MultiJoinInput represents the input tables, key columns and additional columns to be used in the multiJoin. """
```

Comment:
`multiJoin` is Java.  We need common syntax on this.  `multijoin`?  `multi-join`?  `multi-table natural join`?